src/Yasumi/Provider/Canada/Quebec.php

```diff
@@ -0,0 +1,117 @@
+     * country or sub-region.
+     */
+    public const ID = 'CA-QC';
+
+    /**
+     * Initialize holidays for Quebec (Canada).
+     *
+     * @throws InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws UnknownLocaleException
+     * @throws \Exception
+     */
+    public function initialize(): void
+    {
+        parent::initialize();
+        
+        $this->timezone = 'America/Toronto';
+        
+        $this->addHoliday($this->saintJeanBaptisteDay($this->year, $this->timezone, $this->locale));
+        $this->calculateNationalPatriotsDay();
+    }
+
+    /**
+     * Saint-Jean-Baptiste Day.
+     *
+     * The Nativity of John the Baptist (or Birth of John the Baptist, or Nativity of the Forerunner) is a Christian
+     * feast day celebrating the birth of John the Baptist, a prophet who foretold the coming of the Messiah in the
+     * person of Jesus, whom he later baptised. The Nativity of John the Baptist on June 24 comes three months after the
+     * celebration on March 25 of the Annunciation, when the angel Gabriel told Mary that her cousin Elizabeth was in
+     * her sixth month of pregnancy.
+     *
+     * @link https://en.wikipedia.org/wiki/Saint-Jean-Baptiste_Day
+     *
+     * @param int $year the year for which St. John's Day need to be created
+     * @param string $timezone the timezone in which St. John's Day is celebrated
+     * @param string $locale the locale for which St. John's Day need to be displayed in.
+     * @param string $type The type of holiday. Use the following constants: TYPE_OFFICIAL, TYPE_OBSERVANCE,
+     *                         TYPE_SEASON, TYPE_BANK or TYPE_OTHER. By default an official holiday is considered.
+     *
+     * @return Holiday
+     *
+     * @throws InvalidDateException
+     * @throws UnknownLocaleException
+     * @throws \InvalidArgumentException
+     * @throws \Exception
+     */
+    public function saintJeanBaptisteDay(
+        int $year,
+        string $timezone,
+        string $locale,
+        string $type = Holiday::TYPE_OFFICIAL
+    ): Holiday {
+        return new Holiday(
+            'saintJeanBaptisteDay',
+            ['en' => 'Saint-Jean-Baptiste Day', 'fr' => 'Fête de la Saint-Jean-Baptiste'],
+            new DateTime("$year-06-24", DateTimeZoneFactory::getDateTimeZone($timezone)),
+            $locale,
+            $type
+        );
+    }
+
+    /**
+     * National Patriot's Day.
+     *
+     * @link https://en.wikipedia.org/wiki/National_Patriots%27_Day
+     *
+     * @throws InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws UnknownLocaleException
+     * @throws \Exception
+     */
+    private function calculateNationalPatriotsDay(): void
+    {
+        if ($this->year < 2003) {
+            return;
+        }
+
+        $this->addHoliday(new Holiday(
+            'nationalPatriotsDay',
+            ['en' => 'National Patriot\'s Day', 'fr' => 'ournée nationale des patriotes'],
```

Comment:
The test for this holiday fails because the name contains a regular apostrophe. We require the use of the typewriter's apostrophe.